### PR TITLE
Add WeatherService test

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

**Review by alesyalis — APPROVED** [review]

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 1]
```diff
@@ -30,6 +30,9 @@ public async Task<Root> GetWeatherAsync(string cityName)
 
                 var result = JsonConvert.DeserializeObject<Root>(weather);
 
+                if(result.cod == 404)
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:33]: HttpStatusses enum

File: ConsoleApp/Tests/Services/WeatherServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        public WeatherServiceTest()
+        {
+            _weatherFixture = new WeatherFixture();
+            _fixture = (Fixture)new Fixture().Customize(new AutoMoqCustomization());
+            _weatherRepositoryMock = _fixture.Freeze<Mock<IWeatherRepository>>();
+            _weatherService = _fixture.Create<WeatherService>();   
+        }
+
+        [Fact]
```
Comment (denja2) [inline ConsoleApp/Tests/Services/WeatherServiceTest.cs:28]: see theory approach

File: ConsoleApp/Tests/Services/WeatherServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+            _weatherRepositoryMock.Setup(x => x.GetWeatherAsync(It.IsAny<string>()))
+                .ReturnsAsync(expected);
+
+            //Act
+            var actualResult = await _weatherService.GetWeatherAsync(expected.name);
+
+            //Assert
+            Assert.Equal(expectedMessage, actualResult);
+        }
```
Comment (denja2) [inline ConsoleApp/Tests/Services/WeatherServiceTest.cs:111]: check when exception thrown from validator and repository

File: ConsoleApp/Tests/Validators/ValidatorTest.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly IValidator<Root> _validator;
+
+        public ValidatorTest()
+        {
+            _fixture = (Fixture)new Fixture().Customize(new AutoMoqCustomization());
+            _validator = _fixture.Create<Validator<Root>>();
+        }
+
+        [Fact]
+        public void Validate_IfEntityIsNull_ValidationIsFailed()
```
Comment (denja2) [inline ConsoleApp/Tests/Validators/ValidatorTest.cs:23]: unify with the next

File: ConsoleApp/Tests/Validators/ValidatorTest.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            //Arrange
+            string cityName = string.Empty;
+
+            //Act
+            void actualResult() => _validator.ValidateCityName(cityName);
+
+            //Assert
+            Assert.Throws<ValidatorException>((actualResult));
+        }
```
Comment (denja2) [inline ConsoleApp/Tests/Validators/ValidatorTest.cs:46]: add success pass

**Review by denja2 — CHANGES_REQUESTED** [review]

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs (hunk 1 above)
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs:33]: done

File: ConsoleApp/Tests/Services/WeatherServiceTest.cs (hunk 2 above)
Comment (Xodoc) [reply ConsoleApp/Tests/Services/WeatherServiceTest.cs:28]: done

File: ConsoleApp/Tests/Validators/ValidatorTest.cs (hunk 5 above)
Comment (Xodoc) [reply ConsoleApp/Tests/Validators/ValidatorTest.cs:46]: done

File: ConsoleApp/Tests/Services/WeatherServiceTest.cs (hunk 3 above)
Comment (Xodoc) [reply ConsoleApp/Tests/Services/WeatherServiceTest.cs:111]: done

**Review by ihhyy — APPROVED** [review]

File: ConsoleApp/Tests/Validators/ValidatorTest.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+            //Act
+            void actualResult() => _validator.Validate(input);
+
+            //Assert
+            Assert.Throws<ValidatorException>((actualResult));
+        }
+
+        [Theory]
+        [InlineData(" ")]
```
Comment (denja2) [inline ConsoleApp/Tests/Validators/ValidatorTest.cs:36]: add case with ""

Comment (Xodoc) [reply ConsoleApp/Tests/Validators/ValidatorTest.cs:36]: done

**Pull request merged**